relayer/provider/cosmos/provider.go

```diff
@@ -1128,72 +1137,81 @@ func (cc *CosmosProvider) relayPacketsFromResultTx(ctx context.Context, src, dst
 			case !rp.pass:
 				rcvPackets = append(rcvPackets, rp)
 			}
-		}
-	}
 
-	// If there is a relayPacket, return it
-	if len(rcvPackets)+len(timeoutPackets) > 0 {
-		return rcvPackets, timeoutPackets, nil
+			// If there is a relayPacket, return it
+			if len(rcvPackets)+len(timeoutPackets) > 0 {
```

Comment:
Strange notation here -- ought to be `if len(rcvPackets) > 0 || len(timeoutPackets) > 0`.